okTurtles/group-income
# #1184 Enable auto page reloading on GI update available

Closes #1184

### Summary of changes:

- A new `process.env.GI_VERSION` variable has been made available to both the backend and the frontend. It is assigned a new value on every fresh build, but not on partial rebuilds. 
- In production mode, the pubsub server sends the backend `GI_VERSION` to the pubsub client as soon as it reconnects. This allows the frontend to emit a `GI_UPDATE_AVAILABLE` event if the versions don't match, which for now results in an automatic page reload.
  Note that this behavior only applies in production.  In development, BrowserSync already reloads the page as necessary, so there would be a conflict.
 - Minification via ESBuild options has been enabled in production. 
 
 ### Additional info:
 - To test these changes locally:
  1. Set `NODE_ENV` to `production` before running `grunt`.
  2. Open the app in the browser using port 8000, not 3000, because we don't want BrowserSync to interfere.
  3. Shutdown the server and run `grunt` again to rebuild everything. This will make the client start trying to reconnect.
  4. Once the server is up again, the client should need less than one more minute to reconnect.
  5. Upon reconnection the browser should log "New GroupIncome version available" and automatically reload the page.
